subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -430,10 +431,21 @@ int lwm2m_delete_obj_inst(u16_t obj_id, u16_t obj_inst_id)
 
 	engine_unregister_obj_inst(obj_inst);
 	obj->instance_count--;
+#ifdef CONFIG_LWM2M_RD_CLIENT_SUPPORT
+	engine_trigger_update();
```

Comment:
I realize that this isn't a change you made, but we probably shouldn't trigger the registration update until after we've called the obj->delete_cb and reset the resources.